CondTools/Hcal/plugins/HcalPFCutsPopConAnalyzer.cc

```diff
@@ -0,0 +1,39 @@
+#include "CondCore/PopCon/interface/PopConAnalyzer.h"
+#include "CondTools/Hcal/interface/HcalPFCutsHandler.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+//typedef popcon::PopConAnalyzer<HcalPFCutsHandler> HcalPFCutsPopConAnalyzer;
+
+class HcalPFCutsPopConAnalyzer : public popcon::PopConAnalyzer<HcalPFCutsHandler> {
+public:
+  typedef HcalPFCutsHandler SourceHandler;
+
+  HcalPFCutsPopConAnalyzer(const edm::ParameterSet& pset)
+      : popcon::PopConAnalyzer<HcalPFCutsHandler>(pset),
+        m_populator(pset),
+        m_source(pset.getParameter<edm::ParameterSet>("Source")),
+        m_tok(esConsumes<HcalPFCuts, HcalPFCutsRcd>()) {}
+
+private:
+  void endJob() override {
+    m_source.initObject(myDBObject);
+    write();
+  }
+
+  void analyze(const edm::Event& ev, const edm::EventSetup& esetup) override {
+    //Using ES to get the data:
+
+    myDBObject = new HcalPFCuts(esetup.getData(m_tok));
```

Comment:
instead of a `new`, could this be done as a unique_ptr?

Reply:
Right but you are adding a new file here, I didnt ask to fix the rest of them (well now that you are bringing that up, maybe you should do that in a dedicated next PR...), I asked to have a new addition to be better. If you are not using the `unique_ptr` you need to make sure that `myDBObject` is deleted (otherwise this will do a memory leak). Can you point me to where that happens in the code?

Reply:
I am not sure if it gets deleted... The pointer is passes to the PopCon system in the "endJob" method, and then PopCon does whatever is necessary to write the corresponding object or collection to the database. So even if there is a memory leak in a sense that the program does not itself delete every object it creates on the heap, all these objects have to be kept alive until the end of the job anyway, and the OS will do the right thing when the program exits.

Reply:
HcalZDCLowGainFractions is a kind of dead end (not sure it was ever used) so may not serve as an example (?)

Reply:
Most of the code in CondTools/Hcal/plugins is quite old, and was written when unique_ptr wasn't even part of the C++ standard. My preference would be to keep the code consistent with the old design, and if we get to fixing things, do it all at once.